client/src/app/pages/migration-targets/migration-targets.tsx

```diff
@@ -124,22 +134,39 @@ export const MigrationTargets: React.FC = () => {
           response.data.id,
         ]);
       }
-    }
 
+      // Make sure the new target's provider is part of the provider's filter so it can be seen
+      if (filterState.filterValues["provider"]) {
+        let newProviderFilter: FilterValue = null;
+
+        const targetProvider = response.data.provider;
+        if (targetProvider) {
```

Comment:
this part looks useful but tricky
1. if there is no `targetProvider` (never?) we should just return
2. `newProviderFilter` may be const

Reply:
If there is no provider (which is permissible on hub), then the only way to see it is to remove the filter by setting the filterState to null.